diff --git a/meson.build b/meson.build
index a56bb1f37c..f238d57c3f 100644
--- a/meson.build
+++ b/meson.build
@@ -1736,6 +1736,11 @@ major = client_h_define.split('|')[0].split('<<')[0].strip('() ')
 minor = client_h_define.split('|')[1].strip('() ')
 client_api_version = major + '.' + minor + '.0'
 
+## fftools-ffi hack
+libfftools_ffi = dependency('fftools-ffi')
+dependencies += libfftools_ffi
+sources += 'fftools-ffi.c'
+
 libmpv = library('mpv', sources, dependencies: dependencies, gnu_symbol_visibility: 'hidden',
                  link_args: cc.get_supported_link_arguments(['-Wl,-Bsymbolic']),
                  version: client_api_version, include_directories: includedir,
 
diff --git a/fftools-ffi.c b/fftools-ffi.c
new file mode 100644
index 0000000..6cd4258
--- /dev/null
+++ b/fftools-ffi.c
@@ -0,0 +1,6 @@
+#include "fftools-ffi/dart_api.h"
+
+void* a = FFToolsFFIInitialize;
+void* b = FFToolsFFIExecuteFFmpeg;
+void* c = FFToolsFFIExecuteFFprobe;
+void* d = FFToolsCancel;
